src/main/resources/frontend/src/network/ServerCommunicator.ts

```diff
@@ -0,0 +1,287 @@
+import { useAuthStore } from '@/stores/auth'
+import { ServerError } from '@/network/ServerError'
+import { useAppConfigStore } from '@/stores/appConfig'
+
+/**
+ * Utility for making authenticated HTTP requests to the server with automatic error handling
+ * and response parsing.
+ *
+ * @example
+ * // GET request expecting a User response
+ * const user = await ServerCommunicator.getRequest<User>('/api/user');
+ *
+ * // POST request with body, not expecting response
+ * await ServerCommunicator.postRequest<void>('/api/logs', { event: 'action' }, false);
+ */
+export const ServerCommunicator = {
+  getRequest: getRequest,
+  getRequestGuaranteed: getRequestGuaranteed,
+  postRequest: postRequest,
+  patchRequest: patchRequest,
+  doUnprocessedRequest: doUnprocessedRequest
+}
+
+/**
+ * Makes a GET request to the specified endpoint with a guaranteed response.
+ * This will not throw an error if the server returns an error, but will print it
+ * to the console.
+ * @template T - The type of the expected response (when expectResponse is true)
+ * @param {string} endpoint - The API endpoint to call
+ * @param {T} errorResponse - The object the method call should return if the server
+ * returns nothing or responds with a non-2XX code
+ * @returns {Promise<T>} Promise that resolves to the response data of type T
+ */
+async function getRequestGuaranteed<T>(endpoint: string, errorResponse: T): Promise<T> {
+  try {
+    return await getRequest<T>(endpoint, true)
+  } catch (e) {
+    return errorResponse
```

Comment:
Great work!

Heads up: I notice that the documentation indicates an error will be logged, but I don't see the `catch` actually logging an error.

Reply:
Yep! This is intentional. The `doRequest` method already logs the error before throwing a `ServerError`, so additional logging is not needed.